### PR TITLE
API1600 support for HypervisorClusterProfile

### PR DESCRIPTION
**Pull request by nabhajit-ray**
### Description
[API1600 support for HypervisorClusterProfile]

### Issues Resolved
[]

### Check List
- [x] New functionality includes testing.
  - [x]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: endpoints-support.md [hunk 1]
```diff
@@ -43,6 +43,12 @@
 |<sub>/rest/fcoe-networks</sub>                                                           | POST     | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
 |<sub>/rest/fcoe-networks/{id}</sub>                                                      | PUT      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
 |<sub>/rest/fcoe-networks/{id}</sub>                                                      | DELETE   | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+:heavy_mulitplication_x:
+|     **Hypervisor Cluster Profile**                                                                                                                  |
```
Comment (AnushaY1916) [inline endpoints-support.md:47]: I see only PUT endpoint is supported for API 800. Is it intentional? If not correct it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 ### Oneview Features supported
 - Enclosure
 - Enclosure Group
+- Hypervisor Cluster Profile
```
Comment (AnushaY1916) [inline CHANGELOG.md:12]: Line number 6 says Hypervisor Manager and Server Certificate resources are supported. But it's not listed below. Task resource doesn't exist. Remove this. Fix theCHANGELOG

File: examples/hypervisor_cluster_profile.tf [hunk 3]
```diff
@@ -0,0 +1,96 @@
+provider "oneview" {
+	ov_username = "Administrator"
+	ov_password = "admin123"
```
Comment (AnushaY1916) [inline examples/hypervisor_cluster_profile.tf:3]: Put placeholders.

File: examples/hypervisor_cluster_profile.tf [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+data "oneview_scope" "scope_obj" {
+        name = "test_scope"
+}
+
+resource "oneview_hypervisor_cluster_profile" "HypervisorClusterProfile"{  
+    "type"="HypervisorClusterProfileV3",
+    "name"="Cluster",
+    "description"="asa",
```
Comment (AnushaY1916) [inline examples/hypervisor_cluster_profile.tf:17]: Give a proper description.

File: examples/hypervisor_cluster_profile.tf [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                                  "virtual_switch_type"="Standard"
+                               },
+    "hypervisor_host_profile_template"={  
+                                  "server_profile_template_uri"="/rest/server-profile-templates/87afd6f4-8b24-4182-8f06-8b5cc5b40786"
+                                  "host_prefix"="Cluster",
+
+     }
+}
+/*
+
```
Comment (AnushaY1916) [inline examples/hypervisor_cluster_profile.tf:34]: Comment saying that this resource body is for update and needs to be uncommented when used.

File: oneview/data_source_hypervisor_cluster_profile.go [hunk 6]
```diff
@@ -0,0 +1,529 @@
+// (C) Copyright 2016 Hewlett Packard Enterprise Development LP
```
Comment (AnushaY1916) [inline oneview/data_source_hypervisor_cluster_profile.go:1]: Change the year

File: oneview/data_source_hypervisor_cluster_profile.go [hunk 7]
```diff
@@ -0,0 +1,529 @@
...
+			},
+
+			"modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (AnushaY1916) [inline oneview/data_source_hypervisor_cluster_profile.go:303]: If you want to support Get by name and get by URI then make name and URI both as optional. Hope this is what you are trying here.

File: examples/hypervisor_cluster_profile.tf [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        "virtual_switch_uplinks"=[{
+            "name"="Mezz 3:1-c",
+            "active"=false,
+            "action"="NONE"
+          }],
+        "action"="NONE",
+        "network_uris"=["/rest/ethernet-networks/8fa842d8-558e-4958-80cf-116bad88e68b"]
+      }],
+}/*
+
```
Comment (AnushaY1916) [inline examples/hypervisor_cluster_profile.tf:86]: As deleting this resource required an option, please provide the example here with a proper explanation of usage.

File: oneview/data_source_hypervisor_cluster_profile_test.go [hunk 9]
```diff
@@ -0,0 +1,38 @@
+// (C) Copyright 2019 Hewlett Packard Enterprise Development LP
```
Comment (AnushaY1916) [inline oneview/data_source_hypervisor_cluster_profile_test.go:1]: Change year

File: oneview/provider.go [hunk 10]
```diff
@@ -74,6 +74,7 @@ func Provider() terraform.ResourceProvider {
 			"oneview_fc_network":                 dataSourceFCNetwork(),
 			"oneview_fcoe_network":               dataSourceFCoENetwork(),
 			"oneview_hypervisor_manager":         dataSourceHypervisorManager(),
+			"oneview_hypervisor_cluster_profile": dataSourceHypervisorClusterProfile(),
```
Comment (AnushaY1916) [inline oneview/provider.go:77]: Maintain alphabetical order.

File: oneview/resource_hypervisor_cluster_profile.go [hunk 11]
```diff
@@ -0,0 +1,1078 @@
+// (C) Copyright 2016 Hewlett Packard Enterprise Development LP
```
Comment (AnushaY1916) [inline oneview/resource_hypervisor_cluster_profile.go:1]: Change year

File: oneview/data_source_hypervisor_cluster_profile.go [hunk 12]
```diff
@@ -0,0 +1,529 @@
...
+	for i, sharedStorageVolume := range hypCP.SharedStorageVolumes {
+		sharedStorageVolumes[i] = sharedStorageVolume
+	}
+	d.Set("shared_storage_volumes", sharedStorageVolumes)
+	d.Set("state", hypCP.State)
+	d.Set("state_reason", hypCP.StateReason)
+	d.Set("status", hypCP.Status)
+	d.Set("type", hypCP.Type)
+	d.Set("uri", hypCP.URI)
+	d.Set("soft_delete", true)
```
Comment (AnushaY1916) [inline oneview/data_source_hypervisor_cluster_profile.go:527]: Schema doesn't have soft_delete attribute. Remove this.

File: .travis.yml [hunk 13]
```diff
@@ -25,7 +25,8 @@ install:
     - mkdir -p $TF_DOWNLOAD_PATH
     - bash <(curl -s https://raw.githubusercontent.com/patrickdappollonio/download-terraform-binaries/master/download_terraform_binaries.sh)
 
-before_script:
+before_script:  
+    - echo -e $UNFORMATTED
```
Comment (AnushaY1916) [inline .travis.yml:29]: Remove debugging lines

Comment (nabhajit-ray) [reply .travis.yml]: done

File: oneview/data_source_hypervisor_cluster_profile.go (hunk 12 above)
Comment (nabhajit-ray) [reply oneview/data_source_hypervisor_cluster_profile.go]: done

File: oneview/resource_hypervisor_cluster_profile.go (hunk 11 above)
Comment (nabhajit-ray) [reply oneview/resource_hypervisor_cluster_profile.go]: done

File: oneview/provider.go (hunk 10 above)
Comment (nabhajit-ray) [reply oneview/provider.go:76]: done

File: oneview/data_source_hypervisor_cluster_profile_test.go (hunk 9 above)
Comment (nabhajit-ray) [reply oneview/data_source_hypervisor_cluster_profile_test.go]: done

File: examples/hypervisor_cluster_profile.tf (hunk 5 above)
Comment (nabhajit-ray) [reply examples/hypervisor_cluster_profile.tf:35]: done

File: examples/hypervisor_cluster_profile.tf (hunk 4 above)
Comment (nabhajit-ray) [reply examples/hypervisor_cluster_profile.tf]: done

File: examples/hypervisor_cluster_profile.tf (hunk 3 above)
Comment (nabhajit-ray) [reply examples/hypervisor_cluster_profile.tf]: done

File: CHANGELOG.md (hunk 2 above)
Comment (nabhajit-ray) [reply CHANGELOG.md:12]: Done

File: endpoints-support.md (hunk 1 above)
Comment (nabhajit-ray) [reply endpoints-support.md:47]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (AnushaY1916) [reply CHANGELOG.md:12]: You don't need to add these entries manually. rebase you branch with the master.

File: oneview/data_source_hypervisor_cluster_profile.go [hunk 14]
```diff
@@ -0,0 +1,528 @@
...
+
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```
Comment (AnushaY1916) [inline oneview/data_source_hypervisor_cluster_profile.go:352]: Document(docs/d) says that uri is mandatory, but in this file name is mentioned as mandatory and uri as optional. Please fix this.

File: .travis.yml [hunk 15]
```diff
@@ -25,7 +25,7 @@ install:
     - mkdir -p $TF_DOWNLOAD_PATH
     - bash <(curl -s https://raw.githubusercontent.com/patrickdappollonio/download-terraform-binaries/master/download_terraform_binaries.sh)
 
-before_script:
+before_script:
```
Comment (AnushaY1916) [inline .travis.yml:28]: Remove the extra space

File: oneview/data_source_hypervisor_cluster_profile.go (hunk 14 above)
Comment (nabhajit-ray) [reply oneview/data_source_hypervisor_cluster_profile.go:352]: Fixed.

File: .travis.yml (hunk 15 above)
Comment (nabhajit-ray) [reply .travis.yml]: done

File: CHANGELOG.md [hunk 16]
```diff
@@ -10,6 +10,8 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 - Deployment Plan
 - Enclosure
 - Enclosure Group
+- Hypervisor Cluster Profile
+- Hypervisor Manager
```
Comment (AnushaY1916) [inline CHANGELOG.md:14]: Remove duplicate entry

File: oneview/resource_hypervisor_cluster_profile.go [hunk 17]
```diff
@@ -0,0 +1,1086 @@
...
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"type": {
+				Type:     schema.TypeString,
```
[the remark this answers is not in the document]
Comment (nabhajit-ray) [reply oneview/resource_hypervisor_cluster_profile.go:388]: hypervisor_type is mandatory . Have made changes to docs and the resource file also

File: examples/hypervisor_cluster_profile.tf [hunk 18]
```diff
@@ -0,0 +1,97 @@
+provider "oneview" {
+	ov_username = "<username>" 
+	ov_password = "<password>"
+	ov_endpoint = "<endpoint>"
+	ov_sslverify = false
+	ov_apiversion = <ov-apiversion>
+	ov_ifmatch = "*"
+}
+
+data "oneview_scope" "scope_obj" {
```
Comment (soodpr) [inline examples/hypervisor_cluster_profile.tf:10]: scope_obj not being used in the resource creation below

File: CHANGELOG.md [hunk 19]
```diff
@@ -10,6 +10,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 - Deployment Plan
 - Enclosure
 - Enclosure Group
+- Hypervisor Cluster Profile
```
Comment (soodpr) [inline CHANGELOG.md:13]: Just a suggestion here: To add new resources, lets add it under #New Resources heading.

File: examples/hypervisor_cluster_profile.tf (hunk 18 above)
Comment (nabhajit-ray) [reply examples/hypervisor_cluster_profile.tf]: removed

File: CHANGELOG.md (hunk 19 above)
Comment (nabhajit-ray) [reply CHANGELOG.md:13]: Added in Notes .

File: CHANGELOG.md [hunk 20]
```diff
@@ -3,13 +3,14 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 
 # [v1.3.0] (unreleased)
 ### Notes
-- This release supports to API 800,1000,1200,1600 for Hypervisor Manager and Server Certificate resources.
+- This release add supports to API 800,1000,1200,1600 for Hypervisor Manager,Hypervisor Cluster Profile and Server Certificate resources.
```
Comment (soodpr) [inline CHANGELOG.md:6]: are we supporting these resources from API800 onwards?

**Pull request merged**